app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  has_many :goals, dependent: :destroy
+  has_many :achievements, dependent: :destroy
+  has_many :meals, dependent: :destroy
+  has_many :reviews
+  has_many :favourite_recipes, dependent: :destroy
+  has_many :recipes, through: :meals
+
+  validates :username, :age, :weight, :height, :gender, presence: true
```

Comment:
remove line 9, this cannot happen in a wechat miniapp, as the user is already logged in before they can do anything, therefore this will cause error